### DISCUSSION TITLE
Gradle Managed Devices on Large Linux Github hosted runners.

### DISCUSSION THREAD

**Discussion by yogurtearl** [community/community#70344] [Actions]
### Select Topic Area

Bug

### Body

This doc:
https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners#additional-features-for-larger-runners

says:

> Hardware acceleration for the Android SDK tools makes running Android tests much faster and consumes fewer minutes. For more information on Android hardware acceleration, see [Configure hardware acceleration for the Android Emulator](https://developer.android.com/studio/run/emulator-acceleration) in the Android Developers documentation.

It would be good to have more details. :) Specifically, this is talking about KVM support on linux, but the doc above doesn't mention KVM.
See here: https://developer.android.com/studio/run/emulator-acceleration#vm-linux

Also, it would be good if GitHub provided a working demo of android emulator acceleration on large linux runners.

The KVM doesn't work out of the box because the user doesn't have write permission to `/dev/kvm` which is required for the Android emulator to use it. 

To fix this you need to execute this in your job before you start the emulator. 
```shell
sudo chown $USER /dev/kvm
```

This should not be needed and should just work out-of-the-box.

**Comment by collinmcneese** [comment]
There is a Blog tied to this, looks like it isn't hooked in to the docs though, wondering if that might be an opportunity to update.  

Had you come across this before?: https://github.blog/changelog/2023-02-23-hardware-accelerated-android-virtualization-on-actions-windows-and-linux-larger-hosted-runners/

The blog post above calls out:

> 
> To make use of this on Linux, Actions users will need to add the runner user to the KVM user group
> 
> ```yaml
>       - name: Enable KVM group perms
>         run: |
>             echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
>             sudo udevadm control --reload-rules
>             sudo udevadm trigger --name-match=kvm
> ```
> 

Something like the above step could be added in as a step to the Actions workflow which you are using, or even included as a step in a reusable workflow or composite action
